returnn/hdf.py

```diff
@@ -75,7 +75,10 @@ def run(self):
         data = self.data
         if isinstance(data, dict):
             instanciate_delayed(data)
-            data = str(data)
+            with open("dataset.config", "wt") as dataset_file:
```

Comment:
If you say it works if we write just `train = ...` in a config file, 
then we can do it..

I was just now a bit hesitant that we have a config file in the work folder that only exists after the job has started. But the alternatives are 
* have a Task "write_config" that does only exist if data is not a path
* make the config an output of the job

I'll leave this comment here for a while and if nobody else got better solutions then we keep it.